src/async/legends.ts

```diff
@@ -19,14 +19,15 @@ const layerLegendSelect = ({ layers, runConfiguration, job }: any) => {
 
 export default (store: any) => {
   // Layers legend
-  resync(store, layerLegendSelect, ({ layers, serviceId, objective, climate, region }, io, dispatch) => {
+  resync(store, layerLegendSelect, ({ layers }, io, dispatch) => {
     if (layers.length) {
-      const legendLayers = layers.filter((layer: any) => layer.displayed === true && layer.type !== 'vector')
+      const legendLayers = layers.filter((layer: any) => typeof config.layers[layer].show === 'boolean' ? config.layers[layer].show : config.layers[layer].show(store.getState()) && config.layers[layer].type !== 'vector')
```

Comment:
I'm getting an error in CSRT and SST when trying to view a variable layer. I've selected "Custom" for step 5 and added variable "MAT" for step 6:

<img width="509" alt="image" src="https://user-images.githubusercontent.com/20808982/197870687-8c2caa42-1d74-462b-af97-5800ad22b5ae.png">

When clicking the eye icon shown in the lower right, I get this error:

`legends.js:23 Uncaught TypeError: Cannot read properties of undefined (reading 'show')`

It looks like `config.layers` has the prefix "variable-" before each of the variable layers:

<img width="516" alt="image" src="https://user-images.githubusercontent.com/20808982/197871273-5d93b5ec-5d24-4bce-99fa-35a94d90dbaf.png">

but the layer it's searching for is just called "MAT", so it doesn't find it.

Also, this line is very long. All JS/TS code should be run through a linter and prettier.

Reply:
This should be fixed now.